Apps/Inc/CarState.h

```diff
@@ -71,16 +93,20 @@ typedef struct {
     switch_states_t SwitchStates;
     blinker_states_t BlinkerStates;
 
-    State CruiseControlEnable;
+    CruiseRegenEnable CREnable;    
     State CruiseControlSet;
+    
+    double RegenBrakeRate;
```

Comment:
We're trying to avoid floating point math so this should be done with fixed point. If the setpoints we have are going to be in 25% increments, maybe keep regenBrakeRate as an integer between 0 and 4, and then calculate the desired velocity as (currentVelocity * regenBrakeRate) >> 2. This math would be slightly different for different setpoint arrangements, but that's the general idea.